### DISCUSSION TITLE
building the bootloader

### DISCUSSION THREAD

**Discussion by 20centcroak** [pyinstaller/pyinstaller#5956] [Help] [answered]
- Python 3.8 64 bits
- Windows 7 64 bits
- vs_buildtools 2019 (vs_buildtools__1866933634.1623851059)

I need to build the bootload to avoir trojan false alarm (hope so...) by McAffee software, as mentioned here https://stackoverflow.com/questions/43777106/program-made-with-pyinstaller-now-seen-as-a-trojan-horse-by-avg

I followed the documentation https://pyinstaller.readthedocs.io/en/stable/bootloader-building.html
I installed vs build tools 2019
When I launch the build:
`python .\waf all -vvv`

things get stuck to "Checking for 'msvc'" : 
![image](https://user-images.githubusercontent.com/23664172/123427242-a8f70780-d5c4-11eb-83aa-2ac79390266f.png)

and I cannot get the expected executables.
A quick look to the config.log file shows that vs build tools are ok:
![image](https://user-images.githubusercontent.com/23664172/123427453-efe4fd00-d5c4-11eb-8f43-7d81b4795377.png)

Could you please provide help either in building the bootloader or avoiding trojan detection by McAfee by any other mean (I cannot deactivate McAfee...)?
Tks

**Comment by rokm** [comment, marked as answer]
I guess if you're stuck at "Checking for 'msvc'" then the problem is either with `waf` or your MSVC installation, and not with directly with PyInstaller or its bootloader. 

You could try manually running the script that appears to be stuck, i.e. `D:\\dev\\pyinstaller\\bootloader\\build\\waf-print-msvc-1.bat` and see what happens (although all that seems to do is call only `vcvarsall.bat`...).

Alternatively, try running a simple `waf`-based hello world example:
```
example
├─ waf # <-- copied from pyinstaller/bootloader/waf
├─ wscript # see below
└─ main.c # see below
```
```python
# wscript
def options(opt):
        opt.load('compiler_c')
def configure(cnf):
        cnf.load('compiler_c')
def build(bld):
        bld(features='c cprogram', source='main.c', target='program')
```
```c
/* main.c */
#include <stdio.h>

int main (int argc, char **argv)
{
    printf("Hello world!\n");
    return  0;
}
```
Try running `python.exe waf configure build`, which should trigger the same check for `msvc` as when building the bootloader.

**Reply by 20centcroak** [reply]
thanks, I tried with your hello world example and I got the same issue, then I will try and reinstall msvc...

**Comment by ErikPlachta** [comment]
Had a similar issue, and I finally figured it out. I wrote out how to do it here https://link.medium.com/eUDVuymtfjb in a step-by-step guide.
